docs/DeveloperGuide.md

```diff
@@ -266,31 +266,154 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+1.  User requests to add a new task into the existing task schedule.
+2.  GrAB3 shows the updated task schedule.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. Task description is empty.
+
+    * 1a1. GrAB3 shows an error message.
+    
+        Use case resumes at step 2.
+
+**UC02: Delete a task**
+
+**Preconditions: task schedule must exist**
+
+**Guarantees: The task specified by the user will be deleted**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list all tasks for the specified week.
+2.  GrAB3 displays a list of tasks for that week.
+3.  User requests to delete a specific task from the list generated.
+4.  GrAB3 shows the updated list after deletion of the task.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 2a. The TaskSchedule is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The specified task does not exist.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. GrAB3 shows an error message.
 
       Use case resumes at step 2.
 
+**UC03: Edit a task**
+
+**Preconditions: Task exists in TaskSchedule**
+
+**Guarantees: Task specified by user gets edited**
+
+**MSS**
+
+1.  User requests to list all tasks for a specific week.
+2.  GrAB3 shows a list of tasks for that week.
+3.  User requests to edit the description of a specific task from that list.
+4.  GrAB3 shows the updated task.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The TaskSchedule is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. GrAB3 shows an error message.
+
+    Use case resumes at step 2.
+    
+**UC04: Add a preset task into a task template**
+
+**Preconditions: Task templates in the list of templates exists**
+
+**Guarantees: A new preset task will be added into the specified task template**
+
+**MSS**
+
+1. User request to list all task templates available.
+2. GrAB3 shows a list of task templates.
+3. User adds a preset task to a task template selected from the list generated.
+4. GrAB3 shows the updated task template.
+
+    Use case ends.
+
+**UC05: Load task template into task schedule**
+
+**Preconditions: There are existing tasks in the task template**
+
+**Guarantees: Task schedule with tasks from the template get added for the user**
+
+**MSS**
+1. User requests to view the list of task templates.
+2. GrAB3 shows a list of task templates.
+3. User selects a task template to load into the task schedule.
+4. GrAB3 shows the updated task schedule.
+
+    Use case ends
+
+### Module Schedule
+
+**UC06: Add a module into the module schedule**
+
+**Preconditions: Module list must exist**
+
+**Guarantees: A new module will be added to the module schedule**
+
+**MSS**
+
+1.  User requests to view all modules available.
+2.  GrAB3 shows the module list.
+3.  User selects module to add into module schedule
```

Comment:
Perhaps we should add an extension for the case when a user inputs a module code that's not in the list? Like what you did for UC07.